subprojects/language-native/src/test/groovy/org/gradle/language/nativeplatform/internal/incremental/IncrementalCompileProcessorTest.groovy

```diff
@@ -92,17 +92,20 @@ class IncrementalCompileProcessorTest extends Specification {
     }
 
     def added(TestFile sourceFile) {
+        virtualFileSystem.invalidateAll()
```

Comment:
Why are these needed now?

Reply:
Since the `TestSnapshotter` didn't do any caching, which the virtual file system does. We could add a `TestVirtualFileSystem` which does not do any caching, though I think it is better to invalidate the virtual file system instead.

Reply:
I don't understand why we'd be using an implementation here that requires explicit invalidation. Having to call `invalidateAll()` looks like catering for some internal implementation detail that this unit test shouldn't be concerned with. If using `TestVirtualFileSystem` we can remove that complexity, I think then it's a better choice.